gate-oauth2/src/main/java/com/netflix/spinnaker/gate/security/oauth2/OAuthUserInfoServiceHelper.java

```diff
@@ -45,98 +45,83 @@
-import org.springframework.boot.autoconfigure.security.oauth2.resource.ResourceServerProperties;
-import org.springframework.boot.autoconfigure.security.oauth2.resource.UserInfoTokenServices;
 import org.springframework.security.authentication.BadCredentialsException;
-import org.springframework.security.core.AuthenticationException;
-import org.springframework.security.oauth2.common.OAuth2AccessToken;
-import org.springframework.security.oauth2.common.exceptions.InvalidTokenException;
-import org.springframework.security.oauth2.provider.OAuth2Authentication;
-import org.springframework.security.oauth2.provider.OAuth2Request;
-import org.springframework.security.oauth2.provider.token.ResourceServerTokenServices;
-import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Component;
 
 /**
- * ResourceServerTokenServices is an interface used to manage access tokens. The
- * UserInfoTokenService object is an implementation of that interface that uses an access token to
- * get the logged in user's data (such as email or profile). We want to customize the Authentication
- * object that is returned to include our custom (Kork) User.
+ * A helper class to handle common user loading logic for both OAuth2 and OIDC authentication. This
+ * class extracts shared code to avoid duplication in OAuth2 and OIDC user loading processes.
  */
+@Component
 @Slf4j
-public class SpinnakerUserInfoTokenServices implements ResourceServerTokenServices {
+public class OAuthUserInfoServiceHelper {
+
+  private OAuth2SsoConfig.UserInfoMapping userInfoMapping;
+
+  private OAuth2SsoConfig.UserInfoRequirements userInfoRequirements;
+
+  private PermissionService permissionService;
+
+  private Front50Service front50Service;
 
-  private final ResourceServerProperties sso;
-  private final UserInfoTokenServices userInfoTokenServices;
-  private final CredentialsService credentialsService;
-  private final OAuth2SsoConfig.UserInfoMapping userInfoMapping;
-  private final OAuth2SsoConfig.UserInfoRequirements userInfoRequirements;
-  private final PermissionService permissionService;
-  private final Front50Service front50Service;
-  private final SpinnakerProviderTokenServices providerTokenServices;
+  private AllowedAccountsSupport allowedAccountsSupport;
 
-  private final AllowedAccountsSupport allowedAccountsSupport;
-  private final FiatClientConfigurationProperties fiatClientConfigurationProperties;
-  private final Registry registry;
+  private FiatClientConfigurationProperties fiatClientConfigurationProperties;
+
+  private Registry registry;
+
+  @Autowired(required = false)
+  private SpinnakerProviderTokenServices providerTokenServices;
```

Comment:
Let's leave the constructor injection for this instead of property injection.  More importantly though, does anything ever provide this bean?  I see the SpinnakerProviderTokenServices interface annotated with `@Component`, but does anything ever implement that interface?

Reply:
Put back the implementation that was deleted accidentally. Also changed the implementation to adapt to new OAuth.

https://github.com/spinnaker/gate/pull/1887/commits/af1c8db074110983aa80a588155e4b81c68de872

Reply:
Still looking to remove the `@Autowired` annotation on this member.

Reply:
We use property injection instead of constructor injection because this bean is only created when a specific property (spring.security.oauth2.client.registration.github.client-id) is in the configuration. Since its availability is conditional, constructor injection wouldn't work as the dependency might not always be available.

Reply:
Constructor injection still works using `Optional< SpinnakerProviderTokenServices>`.

Reply:
Done
https://github.com/spinnaker/gate/pull/1887/commits/3efc6add15f73b5ac57b45aae93bcb4fc749d6f9